### DISCUSSION TITLE
Apple's Hand-free unlock with UWB locks (WWDC 2024)

### DISCUSSION THREAD

**Discussion by tokamac** [kormax/apple-home-key-reader#35] [Ideas]
At WWDC 2024, Apple announced this new feature for Home Key: **Hands-free unlock with UWB locks**

Basically, until now, Home Key was able to unlock the door by swiping an iPhone or Apple Watch a few millimeters from an NFC reader (that's your implementation).
The new feature extends this possibility to _automatically unlock the door remotely_ a few feet away when you come back home, without even having to reach for your phone, which can stay in your pocket, jacket or bag. It uses UWB.
This is the exact same evolution as Car Key especially in BMWs, which first used NFC near the car door (BMW Digital Key) and now uses UWB at a distance (BMW Digital Key Plus).

I understand perfectly that the PN532 module you are currently using for this project only has NFC and not UWB. So the first preliminary question before even considering any development would be: 
Are there on the market today any cheap DIY modules with UWB, possibly with NFC as well?

![homekey](https://github.com/kormax/apple-home-key-reader/assets/3685009/14e9d7c9-f8a0-4c5d-a382-10793f7599f1)

**Comment by kormax** [comment]
I've actually mentioned info about UWB in my reverse-engineering repository even before WWDC.

In short: UWB is feature of Aliro protocol, which is a modified version of HomeKey protocol developed by CSA, the same org we have to thank for bringing us Matter.

At this moment in IOS18 beta, Aliro is mentioned a couple of times throughout the OS, but is unavailable for use. So we don't have an opportunity to play around with it currently.

When Aliro becomes available, and spec is either published or reverse-engineered - only then we'll be able to build UWB-based locks like presented.

I assume that any off-the-shelf hardware will be available for use, like the Qorvo UWB module, which costs about 50$.
